packages/babel-traverse/test/removal.js

```diff
@@ -45,4 +45,22 @@ describe("removal", function () {
 
     expect(generate(ast).code).toBe("");
   });
+
+  describe("within an IfStatement", function () {
+    it("does not make consequent null", function () {
+      const rootPath = getPath("if (x) foo(); else bar();");
+      const ifPath = rootPath.get("body.0");
+      ifPath.get("consequent").remove();
+
+      expect(ifPath.get("consequent").node).not.toBeNull();
```

Comment:
```suggestion
      expect(ifPath.get("consequent").type).toBe("BlockStatement");
```

Reply:
This was what I wrote at first, but I got the impression the reason for the hook was to avoid setting a non-nullable field to null.  (The more specific check would definitely make sense for something like `TryStatement`, where a block is mandated by the syntax.)

You know the codebase better than I, though, so I'm happy to yield to whatever your preference is. :)

Reply:
> the reason for the hook was to avoid setting a non-nullable field to null

Exactly, and the new test should clearly specifies _how_ we avoid setting `consequent` to `null`. Or you can even put a stricter one
```js
expect(generate(ast).code).toBe("if (x) {} else bar();");
```

If someone revises `path.replaceWith(EmptyBlockStatement)` to `path.replaceWith(UndefinedNode)`, this test will catch such changes.